index.js

```diff
@@ -39,6 +39,10 @@ Client.prototype.search = function (base, options, controls) {
 
   return new Promise(function (resolve, reject) {
     var searchCallback = function (err, result) {
+      if (result === undefined) {
```

Comment:
Is there a difference for checking for no result, vs checking for the presence of an error?

Reply:
Check line 59 - there at least some cases where `result` provides an `error` event, in which case the `err` value is consulted. In this case, there is no result.